sql/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -397,20 +399,22 @@ public Symbol visitReference(Reference symbol, Context context) {
         public Symbol visitFunction(Function function, Context context) {
 
             String functionName = function.info().ident().name();
+            List<Symbol> arguments = function.arguments();
             if (functionName.equals(EqOperator.NAME)) {
-                if (function.arguments().get(0) instanceof Reference) {
+                if (arguments.get(0) instanceof Reference &&
+                    SymbolVisitors.any(Symbols.IS_COLUMN, arguments.get(1)) == false) {
```

Comment:
I think this is too imperative. I'd rather have this extracted in a method

Reply:
After a short discussion we decided to revert the method extraction as the name would be complex to be accurate, like: `leftSideIsReferenceAndRightSideHasNoReference`.

Reply:
@mfussenegger @matriv I think the `visitFunction` method bears a load cognitive load when reading it. And the ABC metric score would out the roof. I think it's important to not make it even more complex as we keep moving the product forward. 

How about we add an isReference method and call that for left side to test it's true and right side is false? So that if statement will become something along the lines of : 

```
if(isRererence(arguments.get(0)) && !isReference(arguments.get(1)))
```

Also, is it important for the right side to not be a reference OR a column?